portfolio/src/main/webapp/script.js

```diff
@@ -34,9 +34,39 @@ async function getContent() {
 }
 
 function getComments(){
-   String json = fetch('/data')
-   response.setContentType("text/html;");//going to priint directly on page
-   response.getWriter().println(json);;
-  }
+   //confirm((new URL(document.location)).searchParams);
+   fetch('/data?user-comment-num='+getUserNum()).then(response => response.json()).then((comments) => {
+       //need to have the ?user-comment-num in order to pass correct params
+    console.log(comments);
+    const messageBoard = document.getElementById('comments-container');
+    comments.forEach((comment) => {
+      messageBoard.appendChild(createCmtEl(comment));
+    })    
+  });
+}
+
+function getUserNum(){
+    //the current document's URL params
+    let searchParams = (new URL(document.location)).searchParams;
+    //that returns a map like structure, so we can get out desired param with .get()
+    let userNum = searchParams.get("user-comment-num");
+    if(userNum == null || userNum.length === 0){
+        return "1";
+    }
+    return userNum;
+}
+
+function createCmtEl(comment) {
+  const comElem = document.createElement('li');
+  comElem.className = 'Comment';
+
+  const nameElem = document.createElement('span');
+  nameElem.innerText = comment.name;
+
+  const txtElem = document.createElement('span');
+  txtElem.innerText = comment.text;
```

Comment:
This would print the comment as: "nameComment" without any spacing between name and the comments. Can we format this to be more user-friendly.

Reply:
Fixed the formatting.